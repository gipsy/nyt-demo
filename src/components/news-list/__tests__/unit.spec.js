import React from 'react'
import renderer from 'react-test-renderer'
import getElementWithContext from 'react-test-context-provider'
import { store } from '@tests/store'

import NewsList from '@components/news-list'

describe('NewsList', () => {
  test('it renders without crashing', () => {
    const element = getElementWithContext({store}, <NewsList />)
    const component = renderer.create(element).toJSON()
    expect(component).toMatchSnapshot()
  })
})
