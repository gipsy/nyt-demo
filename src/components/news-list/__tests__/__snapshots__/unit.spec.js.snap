// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsList /> should match snapshot 1`] = `
.c1 {
  display: block;
}

.c4 {
  display: block;
}

.c3 {
  display: block;
}

.c6 {
  font-family: 'Helvetica',sans-serif;
  color: #ccc;
  font-weight: 100;
  font-size: 12px;
}

.c2 {
  font-family: 'Libre Franklin',sans-serif;
  text-transform: uppercase;
  font-size: 10px;
  font-weight: 700;
  color: rgb(18,18,18);
  display: block;
}

.c5 {
  font-family: 'Noto Serif KR',sans-serif;
  font-size: 16px;
  font-weight: bold;
}

.c0 {
  display: block;
}

<div>
  <div
    class="ant-list c0 ant-list-vertical ant-list-lg ant-list-split ant-list-something-after-last-item"
  >
    <div
      class="ant-list-pagination"
    >
      <ul
        class="ant-pagination "
        unselectable="unselectable"
      >
        <li
          class="ant-pagination-total-text"
        >
          Total 36208 items
        </li>
        <li
          aria-disabled="true"
          class="ant-pagination-disabled ant-pagination-prev"
          title="Previous Page"
        >
          <a
            class="ant-pagination-item-link"
          >
            <i
              class="anticon anticon-left"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="left"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </i>
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          tabindex="0"
          title="1"
        >
          <a>
            1
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-2"
          tabindex="0"
          title="2"
        >
          <a>
            2
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-3"
          tabindex="0"
          title="3"
        >
          <a>
            3
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-4"
          tabindex="0"
          title="4"
        >
          <a>
            4
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-5 ant-pagination-item-before-jump-next"
          tabindex="0"
          title="5"
        >
          <a>
            5
          </a>
        </li>
        <li
          class="ant-pagination-jump-next ant-pagination-jump-next-custom-icon"
          tabindex="0"
          title="Next 5 Pages"
        >
          <a
            class="ant-pagination-item-link"
          >
            <div
              class="ant-pagination-item-container"
            >
              <i
                class="anticon anticon-double-right ant-pagination-item-link-icon"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="double-right"
                  fill="currentColor"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M533.2 492.3L277.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H188c-6.7 0-10.4 7.7-6.3 12.9L447.1 512 181.7 851.1A7.98 7.98 0 0 0 188 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5zm304 0L581.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H492c-6.7 0-10.4 7.7-6.3 12.9L751.1 512 485.7 851.1A7.98 7.98 0 0 0 492 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5z"
                  />
                </svg>
              </i>
              <span
                class="ant-pagination-item-ellipsis"
              >
                •••
              </span>
            </div>
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-3621"
          tabindex="0"
          title="3621"
        >
          <a>
            3621
          </a>
        </li>
        <li
          aria-disabled="false"
          class=" ant-pagination-next"
          tabindex="0"
          title="Next Page"
        >
          <a
            class="ant-pagination-item-link"
          >
            <i
              class="anticon anticon-right"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="right"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                />
              </svg>
            </i>
          </a>
        </li>
      </ul>
    </div>
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-list-item c1"
        >
          <div
            class="ant-list-item-extra-wrap"
          >
            <div
              class="ant-list-item-main"
            >
              <div
                class="ant-list-item-content ant-list-item-content-single"
              >
                <p
                  class="c2"
                >
                  The New York Times
                </p>
                <div
                  class="ant-list-item-meta c3"
                >
                  <div
                    class="ant-list-item-meta-avatar"
                  >
                    <span
                      class="ant-avatar c4 ant-avatar-circle ant-avatar-image"
                    >
                      <img
                        src="https://www.randjsc.com/wp-content/uploads/2015/05/NYT-logo-square-150x150.jpg"
                      />
                    </span>
                  </div>
                  <div
                    class="ant-list-item-meta-content"
                  >
                    <h4
                      class="ant-list-item-meta-title"
                    >
                      <a
                        class="c5"
                        href="https://www.nytimes.com/2018/09/21/travel/singapore-travel-budget.html"
                      >
                        Singapore, for Those Who Aren’t Crazy Rich
                      </a>
                    </h4>
                    <div
                      class="ant-list-item-meta-description"
                    >
                      In the hit movie “Crazy Rich Asians,” the country looks luxe. But it’s easy to have a great time (and eat memorably well) without spending much.
                    </div>
                  </div>
                </div>
                <time
                  class="c6"
                >
                  11 days ago
                </time>
              </div>
            </div>
            <div
              class="ant-list-item-extra"
            >
              <img
                alt="logo"
                src="https://www.randjsc.com/wp-content/uploads/2015/05/NYT-logo-square-150x150.jpg"
                width="75"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-list-pagination"
    >
      <ul
        class="ant-pagination "
        unselectable="unselectable"
      >
        <li
          class="ant-pagination-total-text"
        >
          Total 36208 items
        </li>
        <li
          aria-disabled="true"
          class="ant-pagination-disabled ant-pagination-prev"
          title="Previous Page"
        >
          <a
            class="ant-pagination-item-link"
          >
            <i
              class="anticon anticon-left"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="left"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </i>
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          tabindex="0"
          title="1"
        >
          <a>
            1
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-2"
          tabindex="0"
          title="2"
        >
          <a>
            2
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-3"
          tabindex="0"
          title="3"
        >
          <a>
            3
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-4"
          tabindex="0"
          title="4"
        >
          <a>
            4
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-5 ant-pagination-item-before-jump-next"
          tabindex="0"
          title="5"
        >
          <a>
            5
          </a>
        </li>
        <li
          class="ant-pagination-jump-next ant-pagination-jump-next-custom-icon"
          tabindex="0"
          title="Next 5 Pages"
        >
          <a
            class="ant-pagination-item-link"
          >
            <div
              class="ant-pagination-item-container"
            >
              <i
                class="anticon anticon-double-right ant-pagination-item-link-icon"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="double-right"
                  fill="currentColor"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M533.2 492.3L277.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H188c-6.7 0-10.4 7.7-6.3 12.9L447.1 512 181.7 851.1A7.98 7.98 0 0 0 188 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5zm304 0L581.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H492c-6.7 0-10.4 7.7-6.3 12.9L751.1 512 485.7 851.1A7.98 7.98 0 0 0 492 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5z"
                  />
                </svg>
              </i>
              <span
                class="ant-pagination-item-ellipsis"
              >
                •••
              </span>
            </div>
          </a>
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-3621"
          tabindex="0"
          title="3621"
        >
          <a>
            3621
          </a>
        </li>
        <li
          aria-disabled="false"
          class=" ant-pagination-next"
          tabindex="0"
          title="Next Page"
        >
          <a
            class="ant-pagination-item-link"
          >
            <i
              class="anticon anticon-right"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="right"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                />
              </svg>
            </i>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
